daphne/src/messages/mod.rs

```diff
@@ -1093,37 +1093,24 @@ pub(crate) fn decode_u32_bytes(bytes: &mut Cursor<&[u8]>) -> Result<Vec<u8>, Cod
     Ok(out)
 }
 
-const URL_SAFE_ENGINE: base64::engine::fast_portable::FastPortable =
-    base64::engine::fast_portable::FastPortable::from(
-        &base64::alphabet::URL_SAFE,
-        base64::engine::fast_portable::NO_PAD,
-    );
-
 /// Encode the input bytes as a URL-safe, base64 string.
-pub fn encode_base64url(input: &[u8]) -> String {
-    base64::encode_engine(input, &URL_SAFE_ENGINE)
+pub fn encode_base64url<T: AsRef<[u8]>>(input: T) -> String {
+    URL_SAFE_NO_PAD.encode(input)
 }
 
 /// Decode the input as a URL-safe, base64 encoding of an `OUT_LEN`-length byte string.
-pub fn decode_base64url<const OUT_LEN: usize>(input: &[u8]) -> Option<[u8; OUT_LEN]> {
+pub fn decode_base64url<T: AsRef<[u8]>, const OUT_LEN: usize>(input: T) -> Option<[u8; OUT_LEN]> {
     let mut bytes = [0; OUT_LEN];
```

Comment:
Tests fail because this buffer is not large enough. This is sort of surprising, since this was OK in base64 version 0.20. See https://github.com/marshallpierce/rust-base64/issues/212.